Library/Homebrew/cmd/test-bot.rb

```diff
@@ -425,6 +425,15 @@ def formula formula_name
       install_args = %w[--verbose]
       install_args << "--build-bottle" unless ARGV.include? "--no-bottle"
       install_args << "--HEAD" if ARGV.include? "--HEAD"
+
+      # Pass --devel or --HEAD to the test-bot in the event formulae lack stable. Supports devel-only/head-only.
+      # head-only should not have devel, but devel-only can have head. Stable can have all three.
+      if formula.devel && !formula.stable?
+        install_args << "--devel"
+      elsif formula.head && !formula.stable? && !formula.devel?
+        install_args << "--HEAD"
+      end
```

Comment:
I think we probably want to instead specialise this behaviour for the `devel-only` or `head-only` taps.